src/VecSim/algorithms/svs/svs_tiered.h

```diff
@@ -198,17 +199,16 @@ template <typename DataType>
 class TieredSVSIndex : public VecSimTieredIndex<DataType, float> {
     using Self = TieredSVSIndex<DataType>;
     using Base = VecSimTieredIndex<DataType, float>;
-    using flat_index_t = BruteForceIndex_Single<DataType, float>;
+    using flat_index_t = BruteForceIndex<DataType, float>;
     using backend_index_t = VecSimIndexAbstract<DataType, float>;
     using svs_index_t = SVSIndexBase;
 
-    // Add: true, Delete: false
-    using journal_record = std::pair<labelType, bool>;
+    // label, oldId, newId
+    using swap_record = std::tuple<labelType, idType, idType>;
     size_t trainingTriggerThreshold;
     size_t updateTriggerThreshold;
     size_t updateJobWaitTime;
-    std::vector<journal_record> journal;
-    std::shared_mutex journal_mutex;
```

Comment:
`journal_mutex` is not needed anymore because there is no any dedicated journal locks happen due to `TakeSnapshot()` removal.

Reply:
Can you explain and document the purpose of the `swaps_journal`?